extensions/wikia/Recirculation/RecirculationHooks.class.php

```diff
@@ -150,11 +150,15 @@ private static function checkIfIsProduction() {
 			$wgWikiaEnvironment !== WIKIA_ENV_STAGING;
 	}
 
-	private static function shouldIndex() {
-		return self::isPrivateOrProduction() || self::isCorrectNameSpace();
+	private static function shoudlNoIndex( $metaDataFromService ) {
```

Comment:
how about naming it `shouldIndex` and negating the returned value? IMO names without negation are easier to read